debian/patches/0080-add-tonemapx-filter.patch

```diff
@@ -0,0 +1,5601 @@
++                                               const struct TonemapIntParams *params);
++
++X86_64_V2 void tonemap_frame_420p10_2_420p10_sse(uint16_t *dsty, uint16_t *dstu, uint16_t *dstv,
++                                                 const uint16_t *srcy, const uint16_t *srcu, const uint16_t *srcv,
++                                                 const int *dstlinesize, const int *srclinesize,
++                                                 int dstdepth, int srcdepth,
++                                                 int width, int height,
++                                                 const struct TonemapIntParams *params);
++
++X86_64_V2 void tonemap_frame_p016_p010_2_nv12_sse(uint8_t *dsty, uint8_t *dstuv,
++                                                  const uint16_t *srcy, const uint16_t *srcuv,
++                                                  const int *dstlinesize, const int *srclinesize,
++                                                  int dstdepth, int srcdepth,
++                                                  int width, int height,
++                                                  const struct TonemapIntParams *params);
++
++X86_64_V2 void tonemap_frame_p016_p010_2_p016_p010_sse(uint16_t *dsty, uint16_t *dstuv,
++                                                       const uint16_t *srcy, const uint16_t *srcuv,
++                                                       const int *dstlinesize, const int *srclinesize,
++                                                       int dstdepth, int srcdepth,
++                                                       int width, int height,
++                                                       const struct TonemapIntParams *params);
++
++#endif //FFMPEG_VF_TONEMAPX_INTRIN_SSE_H
+Index: FFmpeg/libavfilter/vf_tonemapx.h
+===================================================================
+--- /dev/null
++++ FFmpeg/libavfilter/vf_tonemapx.h
+@@ -0,0 +1,86 @@
++/*
++ * This file is part of FFmpeg.
++ *
++ * FFmpeg is free software; you can redistribute it and/or
++ * modify it under the terms of the GNU Lesser General Public
++ * License as published by the Free Software Foundation; either
++ * version 2.1 of the License, or (at your option) any later version.
++ *
++ * FFmpeg is distributed in the hope that it will be useful,
++ * but WITHOUT ANY WARRANTY; without even the implied warranty of
++ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
++ * Lesser General Public License for more details.
++ *
++ * You should have received a copy of the GNU Lesser General Public
++ * License along with FFmpeg; if not, write to the Free Software
++ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
++ */
++
++#ifndef FFMPEG_VF_TONEMAPX_H
++#define FFMPEG_VF_TONEMAPX_H
++
++#include "colorspace.h"
++
++#ifdef REFERENCE_WHITE
++#undef REFERENCE_WHITE
++#endif
++#define REFERENCE_WHITE 203.0f
```

Comment:
IIRC the `REFERENCE_WHITE_ALT=203.0f` is only used by bt2390.
Other method such as hable and reinhard use `REFERENCE_WHITE=100.0f`.

Reply:
The CPU implementation of gable and Reinhard does not use reference white. It is doing it normalized with 1.0f